### PR TITLE
Add attribute visitor for ops S-X

### PR DESCRIPTION
**Pull request by tsocha**
list of affected ops:
- Selu op
- ShapeOf op
- ShuffleChannels op
- Sign op
- Sin op
- Sinh op
- Softmax op
- Sqrt op
- SpaceToDepth op
- Split op
- SquaredDifference op
- Squeeze op
- StridedSlice op
- Tan op
- Tanh op
- Tile op
- TopK op
- Transpose op
- Unsqueeze op
- VariadicSplit op
- Xor op

### CODE REVIEW

File: src/ngraph/op/fused/shuffle_channels.cpp [hunk 1]
```diff
@@ -15,6 +15,7 @@
 //*****************************************************************************
 
 #include "ngraph/op/fused/shuffle_channels.hpp"
+#include "ngraph/attribute_visitor.hpp"
 #include "ngraph/builder/reshape.hpp"
```
Comment (diyessi) [inline src/ngraph/op/fused/shuffle_channels.cpp]: Style checker won't like this order

File: src/ngraph/op/fused/space_to_depth.cpp [hunk 2]
```diff
@@ -41,6 +42,13 @@ op::SpaceToDepth::SpaceToDepth(const Output<Node>& data, const std::string& mode
 {
 }
 
+bool ngraph::op::v0::SpaceToDepth::visit_attributes(AttributeVisitor& visitor)
+{
+    visitor.on_attribute("blocksize", m_blocksize);
```
Comment (ilyachur) [inline src/ngraph/op/fused/space_to_depth.cpp]: Incorrect name of attribute! Please change it to `block_size`

File: src/ngraph/op/fused/squared_difference.cpp [hunk 3]
```diff
@@ -35,6 +36,12 @@ op::SquaredDifference::SquaredDifference(const Output<Node>& x1,
     constructor_validate_and_infer_types();
 }
 
+bool ngraph::op::v0::SquaredDifference::visit_attributes(AttributeVisitor& visitor)
+{
+    visitor.on_attribute("autobroadcast", m_autobroadcast);
```
Comment (ilyachur) [inline src/ngraph/op/fused/squared_difference.cpp]: `auto_broadcast`

File: src/ngraph/op/topk.cpp [hunk 4]
```diff
@@ -257,6 +258,15 @@ op::v1::TopK::TopK(const Output<Node>& data,
     constructor_validate_and_infer_types();
 }
 
+bool ngraph::op::v1::TopK::visit_attributes(AttributeVisitor& visitor)
+{
+    visitor.on_attribute("axis", m_axis);
+    visitor.on_attribute("mode", m_mode);
+    visitor.on_attribute("sort", m_sort);
+    visitor.on_attribute("index_element_type", m_index_element_type);
```
Comment (ilyachur) [inline src/ngraph/op/topk.cpp]: `index_element_type` doesn't exist in the spec.

File: src/ngraph/op/xor.cpp [hunk 5]
```diff
@@ -35,6 +35,11 @@ shared_ptr<Node> op::v1::LogicalXor::copy_with_new_args(const NodeVector& new_ar
     return make_shared<v1::LogicalXor>(new_args.at(0), new_args.at(1), this->get_autob());
 }
 
+bool ngraph::op::v1::LogicalXor::visit_attributes(AttributeVisitor& visitor)
+{
+    return true;
```
Comment (ilyachur) [inline src/ngraph/op/xor.cpp]: `auto_broadcast`?

File: src/ngraph/op/fused/space_to_depth.cpp (hunk 2 above)
Comment (tsocha) [reply src/ngraph/op/fused/space_to_depth.cpp]: Done

File: src/ngraph/op/fused/squared_difference.cpp (hunk 3 above)
Comment (tsocha) [reply src/ngraph/op/fused/squared_difference.cpp]: done

File: src/ngraph/op/xor.cpp (hunk 5 above)
Comment (tsocha) [reply src/ngraph/op/xor.cpp]: Done

File: src/ngraph/op/tensor_iterator.cpp [hunk 6]
```diff
@@ -38,6 +38,11 @@ op::TensorIterator::TensorIterator(const OutputVector& values)
 {
 }
 
+bool ngraph::op::TensorIterator::visit_attributes(AttributeVisitor& visitor)
+{
+    return true;
```
Comment (ilyachur) [inline src/ngraph/op/tensor_iterator.cpp]: How should I set/get Body of TI? If I create a TI as an abstract layer how can I create it?

Comment (tsocha) [reply src/ngraph/op/tensor_iterator.cpp]: Since this operator is misalignment with the spec I will remove it from this PR.
TensorIterator should be adjusted in another PR.

**Comment by aslepko** [comment]
TESTNOW_DLDT_E2E

**Comment by ilyachur** [comment]
Could you please add unit tests?

**Comment by tsocha** [comment]
> @ilyachur 
> 
> Could you please add unit tests?

Done.

File: test/attributes.cpp [hunk 7]
```diff
@@ -256,3 +256,136 @@ TEST(attributes, user_op)
...
+{
+    FactoryRegistry<Node>::get().register_factory<opset1::ShuffleChannels>();
+    auto data = make_shared<op::Parameter>(element::i32, Shape{200});
+    auto axis = 0;
+    auto groups = 2;
+    auto shuffle_channels = make_shared<opset1::ShuffleChannels>(data, axis, groups);
+    NodeBuilder builder(shuffle_channels);
+    auto g_shuffle_channels = as_type_ptr<opset1::ShuffleChannels>(builder.create());
+
+    EXPECT_EQ(g_shuffle_channels->get_axis(), shuffle_channels->get_axis());
```
Comment (ilyachur) [inline test/attributes.cpp]: @diyessi @tsocha Can we add several tests which create operation with using visitors but not all attributes are defined?

For example (for this case) what happened if axis is not defined? Can we detect that axis wasn't defined?

These tests can be implemented in the separate PR.

**Comment by ilyachur** [comment]
In general LGTM. But I have an proposal about test cases.

@diyessi @tsocha Can we add several tests which create operation with using visitors but not all attributes are defined?

For example (for this case) what happened if axis is not defined? Can we detect that axis wasn't defined?

These tests can be implemented in the separate PR.
